File: app/src/main/java/it/droidcon/testingdaggerrxjava/userlist/UserListActivity.java
package it.droidcon.testingdaggerrxjava.userlist;

import android.os.Bundle;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.widget.TextView;

import javax.inject.Inject;

import it.droidcon.testingdaggerrxjava.MyApp;
import it.droidcon.testingdaggerrxjava.R;

import static android.support.design.widget.Snackbar.LENGTH_LONG;

public class UserListActivity extends AppCompatActivity {

    @Inject UserListPresenter presenter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ((MyApp) getApplicationContext()).getComponent()
                .userListComponent(new UserListModule(this)).inject(this);
        setContentView(R.layout.activity_main);
        presenter.reloadUserList();
    }

    public void updateText(String s) {
        ((TextView) findViewById(R.id.text)).setText(s);
    }

    public void showError(Throwable t) {
        Snackbar.make(findViewById(android.R.id.content),
                t.getMessage(), LENGTH_LONG).show();
    }
}

File: app/src/main/java/it/droidcon/testingdaggerrxjava/userlist/UserListActivity.kt
package it.droidcon.testingdaggerrxjava.userlist

import android.os.Bundle
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import it.droidcon.testingdaggerrxjava.R
import it.droidcon.testingdaggerrxjava.component
import javax.inject.Inject

class UserListActivity : AppCompatActivity() {

    @Inject lateinit var presenter: UserListPresenter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        component.userListComponent(UserListModule(this)).inject(this)
        setContentView(R.layout.activity_main)
        presenter.reloadUserList()
    }

    fun updateText(s: String) {
        findViewById<TextView>(R.id.text).text = s
    }

    fun showError(t: Throwable) {
        Toast.makeText(this, t.message, Toast.LENGTH_SHORT).show()
    }
}
